hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -72,8 +71,11 @@
   public static final String PRECOMBINE_FIELD_PROP = "hoodie.datasource.write.precombine.field";
   public static final String WRITE_PAYLOAD_CLASS = "hoodie.datasource.write.payload.class";
   public static final String DEFAULT_WRITE_PAYLOAD_CLASS = OverwriteWithLatestAvroPayload.class.getName();
+
   public static final String KEYGENERATOR_CLASS_PROP = "hoodie.datasource.write.keygenerator.class";
-  public static final String DEFAULT_KEYGENERATOR_CLASS = SimpleAvroKeyGenerator.class.getName();
+  public static final String KEYGENERATOR_TYPE_PROP = "hoodie.datasource.write.keygenerator.type";
+  public static final String DEFAULT_KEYGENERATOR_TYPE = "simple";
```

Comment:
Can you set Default as KeyGeneratorType.SIMPLE.name() rather than hardcoding. also, lets standardize if we need to lowercase or leave it as is. 
@n3nash : is there any guideline in general wrt using enums as vals in configs. do we lower case it or expect users to set enum vals as is(UPPERCASE) ?

Reply:
> Can you set Default as KeyGeneratorType.SIMPLE.name() rather than hardcoding. also, lets standardize if we need to lowercase or leave it as is.
> @n3nash : is there any guideline in general wrt using enums as vals in configs. do we lower case it or expect users to set enum vals as is(UPPERCASE) ?

done

Reply:
We always let the enum.name() be the driver of the string, we don't have cases where we hand write the name/string.